src/native/libs/System.Native/pal_uid.c

```diff
@@ -237,3 +238,83 @@ int32_t SystemNative_GetGroups(int32_t ngroups, uint32_t* groups)
 
     return getgroups(ngroups, groups);
 }
+
+char* SystemNative_GetGroupName(uint32_t gid)
+{
+#if HAVE_GETGRGID_R
+    size_t bufferLength = 512;
+    while (1)
+    {
+        char *buffer = (char*)malloc(bufferLength);
+        if (buffer == NULL)
+            return NULL;
+
+        struct group gr;
+        struct group* result;
+        if (getgrgid_r(gid, &gr, buffer, bufferLength, &result) == 0)
+        {
+            if (result == NULL)
+            {
+                errno = ENOENT;
+                free(buffer);
+                return NULL;
+            }
+            else
+            {
+                char* name = strdup(gr.gr_name);
+                free(buffer);
+                return name;
+            }
+        }
+
+        free(buffer);
+        size_t tmpBufferLength;
+        if (errno != ERANGE || !multiply_s(bufferLength, (size_t)2, &tmpBufferLength))
+        {
+            return NULL;
+        }
+        bufferLength = tmpBufferLength;
+    }
+#else
+    (void)gid;
```

Comment:
Just curious, why the cast to void?

Reply:
It suppresses warnings about the parameter being unused.